test/unit/resources/legacy_gelamtest.js

```diff
@@ -0,0 +1,351 @@
+ *      test integrity.
+ *
+ * 5. Most T.{setup,check,...} functions are mapped to T.action.
+ *
+ * 6. This test file imports all of the old test helpers and hooks them up.
+ *
+ * Apart from coalescing some methods as described above, most existing tests
+ * remained largely unchanged, and continue to pass and detect failures as
+ * expected. All LegacyGelamTests can be viewed with the new logic-inspector
+ * alongside new tests.
+ */
+define(function(require) {
+
+  var logic = require('logic');
+  var equal = require('equal');
+  var GelamTest = require('./gelamtest');
+
+  var $th_main = require('./th_main'),
+      $th_fake_servers = require('tests/resources/th_fake_servers'),
+      $th_contacts = require('tests/resources/th_contacts'),
+      $th_devicestorage = require('tests/resources/th_devicestorage');
+
+  var TEST_HELPERS = [
+    $th_main.TESTHELPER,
+    $th_fake_servers.TESTHELPER,
+    $th_contacts.TESTHELPER,
+    $th_devicestorage.TESTHELPER
+  ];
+
+  function Actor(T, RT, ns, name) {
+    if (!ns) {
+      ns = 'LazyActor';
+    }
+    this.T = T;
+    this.RT = RT;
+    this.__name = name;
+    this.ns = ns;
+    logic.defineScope(this, ns);
+    this._resolveStep = null;
+  }
+
+  Actor.currentStepActors = null;
+
+  Actor.prototype = {
+
+    toString: function() {
+      return '[' + this.ns + (this.__name ? ' ' + this.__name : '') + ']';
+    },
+
+    // Internal, you don't have to call this!
+    ensureInStep: function() {
+      if (!Actor.currentStepActors) {
+        throw new Error('Actor ' + this + ' is not in a step!');
+      }
+      if (!Actor.currentStepActors.has(this)) {
+        this.setMatchingEnabled = false;
+        this.unmetExpectations = [];
+        this.notExpectations = [];
+        Actor.currentStepActors.add(this);
+      }
+    },
+
+    expectNot: function(type, details) {
+      this.ensureInStep();
+      var expectation = { not: true, type: type, details: details };
+      logic(this, 'expectNot', expectation);
+      this.notExpectations.push(expectation);
+    },
+
+    expect: function(type, details) {
+      this.ensureInStep();
+      var expectation = { type: type, details: details };
+      logic(this, 'expect', expectation);
+      this.unmetExpectations.push(expectation);
+    },
+
+    error: function(msg) {
+      this.ensureInStep();
+      this.unmetExpectations.push(msg);
+    },
+
+    useSetMatching: function() {
+      this.ensureInStep();
+      this.setMatchingEnabled = true;
+    },
+
+    log: function(name, details) {
+      logic(this, name, details);
+    },
+
+    handleEvent: function(event) {
+      if (event.scope.namespace !== this.ns) {
+        return;
+      }
+
+      for (var i = 0; i < this.notExpectations.length; i++) {
+        var expectation = this.notExpectations[i];
+        if (event.matches(expectation.type, expectation.details)) {
+          this.unmetExpectations.push(expectation);
```

Comment:
It seems really suspicious that notExpectations turn into unmetExpectations.  It seems possible for code gone wild to actually pass in this case if the not situation matches twice.  (Because this only actually generates an error when the timeout occurs.)  Given how few instances there were of expectNothing in the code-base, this does seem like a reasonable shortcut to take, but I think a comment like "// XXX hack to support expectNothing idiom is not fool-proof, but there are almost no call-sites.  All new code should use failIfMatched in logic."

We probably also want to do something like `logic(this, 'expectNotSadness', ...)`.

It may also just be possible to fix the correctness issue by ensuring that event.matches can't possibly trigger once you put the thing on the unmetExpectations list.